pandas/tests/series/indexing/test_datetime.py

```diff
@@ -667,3 +667,15 @@ def test_setitem_tuple_with_datetimetz():
     result[(0, 1)] = np.nan
     expected.iloc[0] = np.nan
     tm.assert_series_equal(result, expected)
+
+
+def test_indexing_regression():
+    # Issue 34860
+    arr = date_range("1/1/2008", "1/1/2009")
+    ser = arr.to_series()["2008"]
+
+    rng = date_range(start="2008-01-01", end="2008-12-31")
+    expected = Series(rng, index=rng)
+
+    tm.assert_series_equal(ser, expected)
+    tm.assert_index_equal(ser.index, expected.index)
```

Comment:
We don't need this `assert_index_equal`, `assert_series_equal` will also test the index.